compiler/rustc_llvm/build.rs

```diff
@@ -235,20 +235,21 @@ fn main() {
     let mut cmd = Command::new(&llvm_config);
     cmd.arg(llvm_link_arg).arg("--libs");
 
-    if !is_crossed {
-        cmd.arg("--system-libs");
-    } else if target.contains("windows-gnu") {
-        println!("cargo:rustc-link-lib=shell32");
-        println!("cargo:rustc-link-lib=uuid");
-    } else if target.contains("netbsd") || target.contains("haiku") || target.contains("darwin") {
-        println!("cargo:rustc-link-lib=z");
-    } else if target.starts_with("arm")
+    if target.starts_with("arm")
         || target.starts_with("mips-")
         || target.starts_with("mipsel-")
         || target.starts_with("powerpc-")
     {
         // 32-bit targets need to link libatomic.
         println!("cargo:rustc-link-lib=atomic");
+        println!("cargo:rustc-link-lib=z");
+    } else if !is_crossed {
+        cmd.arg("--system-libs");
```

Comment:
This doesn't seem right to "demote" `--system-libs`, because if that would list anything, we probably ought to be using those. Maybe we should keep that up front, then *in addition* add extra libraries we think we want for specific targets. (i.e. remove `else if` -- `if !is_crossed {..}`, then `if target...`)

Reply:
Corrected.